Cython/Compiler/ExprNodes.py

```diff
@@ -7012,29 +7012,11 @@ def analyse_as_type_attribute(self, env):
                             return None
                         ubcm_entry = entry
                     else:
-                        # Create a temporary entry describing the C method
-                        # as an ordinary function.
-                        if entry.func_cname and not hasattr(entry.type, 'op_arg_struct'):
-                            cname = entry.func_cname
-                            if entry.type.is_static_method or (
-                                    env.parent_scope and env.parent_scope.is_cpp_class_scope):
-                                ctype = entry.type
-                            elif type.is_cpp_class:
-                                error(self.pos, "%s not a static member of %s" % (entry.name, type))
-                                ctype = PyrexTypes.error_type
-                            else:
-                                # Fix self type.
-                                ctype = copy.copy(entry.type)
-                                ctype.args = ctype.args[:]
-                                ctype.args[0] = PyrexTypes.CFuncTypeArg('self', type, 'self', None)
-                        else:
-                            cname = "%s->%s" % (type.vtabptr_cname, entry.cname)
-                            ctype = entry.type
-                        ubcm_entry = Symtab.Entry(entry.name, cname, ctype)
-                        ubcm_entry.is_cfunction = 1
-                        ubcm_entry.func_cname = entry.func_cname
-                        ubcm_entry.is_unbound_cmethod = 1
-                        ubcm_entry.scope = entry.scope
+                        ubcm_entry = self._create_ubcm_entry(entry, env)
+                        ubcm_entry.overloaded_alternatives = [
+                            self._create_ubcm_entry(overloaded_alternative, env) for overloaded_alternative in
+                            entry.overloaded_alternatives
```

Comment:
I find this more readable since it visibly separates the target item from the loop that creates it. Especially for complex item code, easily spotting the `for` is crucial.
```suggestion
                            self._create_ubcm_entry(overloaded_alternative, env)
                            for overloaded_alternative in entry.overloaded_alternatives
```